wrapper/CSharp/wolfCrypt-Test/wolfCrypt-Test.cs

```diff
@@ -648,6 +648,289 @@ private static void curve25519_test()
+                ret = wolfcrypt.MlKemEncapsulate(keyA, out cipherText, out sharedSecretA);
+                if (ret != 0)
+                {
+                    Console.Error.WriteLine("Failed to encapsulate.");
+                }
+            }
+            if (ret == 0)
+            {
+                Console.WriteLine("ML-KEM Encapsulation test passed.");
+            }
+            
+            /* Decode */
+            if (ret == 0)
+            {
+                Console.WriteLine("Testing ML-KEM Decode...");
+                ret = wolfcrypt.MlKemDecodePrivateKey(keyB, privA);
+                if (ret != 0)
+                {
+                    Console.Error.WriteLine("Failed to decode private key of A.");
+                }
+            }
+            if (ret == 0)
+            {
+                ret = wolfcrypt.MlKemDecodePublicKey(keyB, pubA);
+                if (ret != 0)
+                {
+                    Console.Error.WriteLine("Failed to decode public key of B.");
+                }
+            }
+            if (ret == 0)
+            {
+                Console.WriteLine("ML-KEM Decode test passed.");
+            }
+
+            /* Decapsulate */
+            if (ret == 0)
+            {
+                Console.WriteLine("Testing ML-KEM Decapsulation...");
+                ret = wolfcrypt.MlKemDecapsulate(keyB, cipherText, out sharedSecretB);
+                if (ret != 0)
+                {
+                    Console.Error.WriteLine("Failed to decapsulate.");
+                }
+            }
+            if (ret == 0)
+            {
+                Console.WriteLine("ML-KEM Decapsulation test passed.");
+            }
+
+            /* DEBUG CODE FOR REVIEW */
+            ret = BAD_FUNC_ARG;
+
+            /* Check */
+            if (ret == 0)
+            {
+                Console.WriteLine("Comparing Shared Secrets...");
+                if (!wolfcrypt.ByteArrayVerify(sharedSecretA, sharedSecretB))
+                {
+                    Console.Error.WriteLine("Shared secrets do not match.");
+                }
+                else
+                {
+                    Console.WriteLine("ML-KEM shared secret match.");
+                }
+            }
+        }
+        catch
+        {
+            // todo:
+            // ........ https://github.com/wolfSSL/wolfssl/pull/9040#discussion_r2386193630
+            // ........ https://github.com/wolfSSL/wolfssl/pull/9040#discussion_r2386194020
+        }
+        finally
+        {
+            /* Cleanup */
+            if (keyA != IntPtr.Zero)
+            {
+                ret = wolfcrypt.MlKemFreeKey(keyA);
+                if (ret != 0)
+                {
+                    Console.Error.WriteLine("Failed to free MlKem key A. Error code: " + ret);
+                }
+            }
+            if (keyB != IntPtr.Zero)
+            {
+                ret = wolfcrypt.MlKemFreeKey(keyB);
```

Comment:
Any prior `ret` error values are lost and overwritten here. Is this as intended?